File: frontend/libs/api-enums/src/lib/common/generation/CommonAspectRatio.ts

// NOTE: These are defined in Rust (as the source of truth) and duplicated in the frontend.
// In the future, we should use code gen (protobufs or similar) to keep the two sides in sync.

export enum CommonAspectRatio {
  Auto = "auto",
  Square = "square",
  WideThreeByTwo = "wide_three_by_two",
  WideFourByThree = "wide_four_by_three",
  WideFiveByFour = "wide_five_by_four",
  WideSixteenByNine = "wide_sixteen_by_nine",
  WideTwentyOneByNine = "wide_twenty_one_by_nine",
  TallTwoByThree = "tall_two_by_three",
  TallThreeByFour = "tall_three_by_four",
  TallFourByFive = "tall_four_by_five",
  TallNineBySixteen = "tall_nine_by_sixteen",
  TallNineByTwentyOne = "tall_nine_by_twenty_one",
  Wide = "wide",
  Tall = "tall",
  Auto2k = "auto_2k",
  Auto3k = "auto_3k",
  Auto4k = "auto_4k",
  SquareHd = "square_hd",
}

File: crates/models/src/enums/common_aspect_ratio.rs
use serde_derive::{Deserialize, Serialize};

/// Aspect ratios a model can be asked for. Not every model supports every
/// value; a model's config lists its own.
#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum CommonAspectRatio {
  /// Let the model decide (e.g. follow the source image when editing).
  Auto,
  Square,
  WideThreeByTwo,
  WideFourByThree,
  WideFiveByFour,
  WideSixteenByNine,
  WideTwentyOneByNine,
  TallTwoByThree,
  TallThreeByFour,
  TallFourByFive,
  TallNineBySixteen,
  TallNineByTwentyOne,
  /// Imprecise semantic values some models map to their own sizes.
  Wide,
  Tall,
  /// Auto aspect at a baked-in resolution.
  #[serde(rename = "auto_2k")]
  Auto2k,
  #[serde(rename = "auto_3k")]
  Auto3k,
  #[serde(rename = "auto_4k")]
  Auto4k,
  /// Square at a baked-in HD resolution.
  SquareHd,
}
